src/main/java/com/google/finapp/SpannerDaoJDBCImpl.java

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.finapp;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.*;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+
+final class SpannerDaoJDBCImpl implements SpannerDaoInterface {
+
+  private final String projectId;
+  private final String databaseId;
+  private final String instanceId;
+  private final String connectionUrl;
+
+  @Inject
+  SpannerDaoJDBCImpl(@ArgsModule.SpannerProjectId String spannerProjectId,
+      @ArgsModule.SpannerInstanceId String spannerInstanceId,
+      @ArgsModule.SpannerDatabaseId String spannerDatabaseId) {
+    this.projectId = spannerProjectId;
+    this.databaseId = spannerDatabaseId;
+    this.instanceId = spannerInstanceId;
+    // use this URL to connect to the emulator
+    this.connectionUrl = String.format(
+        "jdbc:cloudspanner://localhost:9010/projects/%s/instances/%s/databases/%s;usePlainText=true",
+        projectId, instanceId, databaseId);
+    // use this URL to connect to Cloud Spanner
+    // this.connectionUrl = String.format(
+    //     "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s",
+    //     projectId, instanceId, databaseId);
+  }
+
+  public void createCustomer(ByteArray customerId, String name, String address) throws SQLException {
+    try (Connection connection = DriverManager.getConnection(this.connectionUrl)) {
+      try (PreparedStatement ps =
+          connection.prepareStatement(
+              "INSERT INTO Customer\n"
+                  + "(CustomerId, Name, Address)\n"
+                  + "VALUES\n"
+                  + "(?, ?, ?)")) {
+        ps.setBytes(1, customerId.toByteArray());
+        ps.setString(2, name);
+        ps.setString(3, address);
+        ps.executeUpdate();
+        System.out.printf("Customer created: %s\n", name);
```

Comment:
+1 to Panos's comment on using logger instead. I'd prefer it if we do it in this PR. 

For some reason, I'm not able to reply on that conversation.

Reply:
I'm going to address logging in a PR immediately following this. I'll have to update both `SpannerDaoImpl` and `SpannerDaoJDBCImpl`, and I want to keep this PR to only modifying/creating `SpannerDaoJDBCImpl`. I'd prefer to setup logging in a single PR. I also want to improve the statements we use for logging. See issue https://github.com/GoogleCloudPlatform/cloud-spanner-samples/issues/11